crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_create_subintent.rs

```diff
@@ -0,0 +1,250 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Create Subintent
+// ==================
+impl SargonOS {
+    /// Creates a Subintent given its discriminator, manifest and expiration.
+    pub async fn create_subintent(
+        &self,
+        intent_discriminator: IntentDiscriminator,
+        subintent_manifest: SubintentManifest,
+        expiration: DappToWalletInteractionSubintentExpiration,
+    ) -> Result<Subintent> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        let current_epoch = gateway_client.current_epoch().await?;
+
+        // Calculate the seconds until the expiration of the subintent.
+        let expiry_time_from_now_in_seconds =
+            self.expiry_time_from_now_in_seconds(expiration);
+        if expiry_time_from_now_in_seconds == 0 {
+            return Err(CommonError::SubintentExpired);
+        }
```

Comment:
This could be moved up, so we avoid making a gateway call is the subintent is expired already